mmv1/products/discoveryengine/Schema.yaml

```diff
@@ -0,0 +1,101 @@
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.dataStores.schemas'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas/{{schema_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas?schemaId={{schema_id}}'
+update_verb: :PATCH
+update_mask: true
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas/{{schema_id}}'
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/schemas/{{schema_id}}',
+  ]
+
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_schema_basic"
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_schema%s", context["random_suffix"])'
+    vars:
+      data_store_id: "data-store-id"
+      schema_id: "schema-id"
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The geographic location where the data store should reside. The value can
+      only be one of "global", "us" and "eu".
+  - !ruby/object:Api::Type::String
+    name: 'dataStoreId'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The unique id of the data store.
+  - !ruby/object:Api::Type::String
+    name: 'schemaId'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The unique id of the schema.
+
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The unique full resource name of the schema. Values are of the format
+      `projects/{project}/locations/{location}/collections/{collection_id}/dataStores/{data_store_id}/schemas/{schema_id}`.
+      This field must be a UTF-8 encoded string with a length limit of 1024
+      characters.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'jsonSchema'
```

Comment:
Can you please add `custom_expand`, `custom_flatten`, `state_func` and `validation` to this field? It should fix the failed test.

Example:
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/healthcare/Hl7V2Store.yaml#L121-L126

Reply:
Thanks, added `custom_expand`, `custom_flatten`, `state_func` and `validation` to this field. The "# whitespace changes" error was resolved. 
But after adding them, an unexpected error `"Invalid value at 'schema' (json_schema), Starting an object on a scalar field"` occurred at the third request `3. Create the resource Schema`.

1. `json_schema` represents a schema in JSON string. The input JSON string has curly braces (`{` and `}`) to hold object ([example](https://source.corp.google.com/piper///depot/google3/cloud/ml/discoveryengine/common/json/schema/predefined/generic.json)). Before and after adding these 4 properties, nothing changed in file `mmv1/templates/terraform/examples/discoveryengine_schema_basic.tf.erb` but the request failed.
2. Referring to this example ([resource](https://github.com/GoogleCloudPlatform/magic-modules/blob/6d3d295685841b27e94564183f31da739990b73a/mmv1/products/firestore/Document.yaml#L96C3-L107C42) & [create test](https://github.com/GoogleCloudPlatform/magic-modules/blob/6d3d295685841b27e94564183f31da739990b73a/mmv1/templates/terraform/examples/firestore_document_basic.tf.erb#L35C17-L35C102)), the usage looked quite the same. I'm not sure whether there should be any corresponding changes for my create test.